airbyte-integrations/connectors/destination-s3-data-lake/src/main/kotlin/io/airbyte/integrations/destination/s3_data_lake/S3DataLakeTableSynchronizer.kt

```diff
@@ -129,6 +129,11 @@ class S3DataLakeTableSynchronizer(
             }
         }
 
+        // 5) Update identifier fields
+        if (diff.identifierFieldsChanged) {
+            update.setIdentifierFields(incomingSchema.identifierFieldNames().toList())
```

Comment:
should we also do `requireColumn(...)` on the fields? and/or make the previous identifier fields no longer required?

Reply:
I did think about it but its not required. When we convert Airbyte schema to iceberg schema in the class `AirbyteTypeToIcebergSchema` we have the check to make the field required if its being marked as an identifier field in the schema. So we dont need that check here again.